docs/sources/http_api/query_history.md

```diff
@@ -85,12 +85,14 @@ Returns a list of queries in the query history that matches the search criteria.
 
 Query parameters:
 
-- **datasourceUid** - Filter the query history for selected data sources. You must specify at least one data source UID. To perform an "AND" filtering with multiple data sources, specify the data source parameter using the following format: `datasourceUid=uid1&datasourceUid=uid2`.
+- **datasourceUid** - Filter the query history for the selected data source. To perform an "AND" filtering with multiple data sources, specify the data source parameter using the following format: `datasourceUid=uid1&datasourceUid=uid2`.
 - **searchString** – Filter the query history based on the content.
 - **sort** - Specify the sorting order. Sorting can be `time-asc` or `time-desc`. The default is `time-desc`.
 - **onlyStarred** - Search for queries that are starred. Defaults to `false`.
 - **page** - Search supports pagination. Specify which page number to return. Use the limit parameter to specify the number of queries per page.
 - **limit** - Limits the number of returned query history items per page. The default is 100 queries per page.
+- **from** - Start time for the query history search as a second UNIX timestamp.
+- **to** - End time for the query history search as a second UNIX timestamp.
```

Comment:
Can we support the same format here as we do for `from`/`to` when querying a data source? https://grafana.com/docs/grafana/latest/http_api/data_source/#query-a-data-source

> from/to – Specifies the time range for the queries. The time can be either epoch timestamps in milliseconds or relative using Grafana time units. For example, now-5m.

Somewhat confusingly, the tooling we have for parsing timestamps is in the legacydata package (I should probably move it out of there, unlike the rest of that package, that's up-to-date and still the correct way to parse time ranges in the backend)

```go
import "github.com/grafana/grafana/pkg/tsdb/legacydata"

func Example() {
tr, err := legacydata.NewDataTimeRange(from, to)
if err != nil …

tr.GetFromAsSecondsEpoch()
tr.GetToAsSecondsEpoch()
}
```

Reply:
That is such a great idea! It is actually going to be very useful, because this way we don't have to calculate time range on frontend and we can just send `now` and `now-1d`. I'll update it. 🙂